joomla/joomla-cms
# Update to String package 1.3

One of the changes in PHP 7 is the platform now allows [Scalar Type Hints](https://wiki.php.net/rfc/scalar_type_hints_v5).  Implementing this change means there are now some new reserved keywords in the language that cannot be used for class names, such as "string".  We have a class named just that in the Framework's String package and have managed its deprecation at that level.

This PR updates the CMS to use the latest String package which includes a new "StringHelper" class to replace the existing "String" class.  JString is modified to extend this directly versus the older class.
### Testing Instructions

JString is used to help create aliases on items when you save them.  So you should be able to create a new item and validate the alias is still processed correctly.
### B/C Implication

As JString, String, and StringHelper are all abstract, there is no practical reason to typehint against this class.  However, in the very off chance someone may be, a JString object will no longer match a typehint for a String class.  The typehint can be updated for StringHelper and everything will work out OK.  I realize this is a fat chance, but it is a B/C break and the options are either this or Joomla 3 will not function on PHP 7.
